### DISCUSSION TITLE
대시보드 공유 기능 (v)

### DISCUSSION THREAD

**Discussion by un0211** [Part3-Team15/taskify#101] [Ideas]
산책하다가 대시보드 공유 기능 만들면 좋을 것 같다는 생각이 들어 작성합니다~~

- 목적: 우리 대시보드를 멤버가 아닌 다른 사람에게 공유한다. 그 사람은 수정이나 멤버관리 권한은 없고 구경만 할 수 있다.
- 방법
  - 우리가 임시계정을 만들고 액세스 토큰을 유출되지 않게 저장한다. (.env ?)
  - 페이지 설정에 모두에게 퍼블릭하게 공개하기 라디오 설정을 만든다.
  - 공개 대시보드인 경우, 자동으로 우리가 만든 임시 계정을 멤버로 추가한다.
    -  반대로 취소하면 자동으로 멤버에서 제외한다.
  - 사용자가 대시보드에 접근하려 할 때, 먼저 그 사람의 액세스 토큰을 본다.
    - 만약 권한이 없다면 우리가 저장한 임시계정의 액세스 토큰을 이용해 한 번 더 시도한다.
    - ! get 함수에만 한 번 더 시도 로직을 추가해서 볼 수만 있게 한다.
    - 만약 퍼블릭 공개라면 임시계정이 멤버일 것이므로 읽을 수 있다.
    - 퍼블릭이 아니라면 임시계정으로도 접근할 수 없다.

롤링 할때는 누구나 볼 수 있었는데, 이건 멤버만 볼 수 있으니 공개 기능이 있으면 어떨까 했습니다.
대시보드 공개할 일이 얼마나 있을진 몰라도, 테스트 해보는 사람이 저희나 친구에게 링크 보여줄 일 있을수도 있으니, 시간  남으면 추가해도 좋을 것 같아요~

**Comment by wayandway** [comment]
혜원님 산책하시고 두뇌 풀가동 되셨군요!!!! ㅋㅋㅋㅋ
공유 기능 좋은 것 같습니다~~ 노션 웹게시랑 비슷한 느낌이네요 ㅎㅎ

**Reply by un0211** [reply]
ㅋㅋㅋㅋㅋ 두뇌 풀가동

**Comment by wjsdncl** [comment]
오 좋은거 같습니다.

**Reply by wjsdncl** [reply]
그런데 이러면 서버작업이 필요하지 않을까요? 공유된 상태를 확인할 방법이 있을까요?

**Reply by un0211** [reply]
그러네요.. 최대한 서버 작업 없이 하려했는데 어렵네요 ㅠ

한 가지 생각은 우리가 만든 임시 계정이 있냐 없냐를 상태로 쓰는 거예요.

- 처음 생성할 때 토글로 선택
  - 퍼블릭 게시 선택하면 멤버에 임시 계정을 가입
- 임시 계정이 멤버로 존재하면 대시보드 수정에서 퍼블릭 o
- 만약 다시 프라이빗으로 바꾼다면 임시 계정 멤버 삭제
- 임시 계정 없으면 퍼블릭 x

다른 방법은 넥스트니까 대시보드 아이디마다 퍼블릭 on/off 만 저장하고 알려주는 API 만드는 거예요!
이정도는 저도 만들 수 있을 것 같아서!!

그래도 기능의 유용성 대비 할일이 너무 많으면 포기해야 할 것 같아요

**Reply by wjsdncl** [reply]
아하 이런 방법이 있군요 산책이 많이 도움되네요 ㅋㅋ
